// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTBlockchainAddress";
option optimize_for = LITE_RUNTIME;

import public "AsymmetricKey.proto";
import public "BlockchainActivity.proto";

message BlockchainAddress
{
    optional uint32 version = 1;
    optional uint32 index = 2;
    optional string label = 3;
    optional string contact = 4;
    optional AsymmetricKey key = 5;
    optional uint64 modified = 6;
    repeated string unconfirmed = 7;
    repeated string confirmed = 8;
}
